### DISCUSSION TITLE
Cannot create a correct "secretKey" for JWE

### DISCUSSION THREAD

**Discussion by Hleb-Haiduk-Software** [panva/jose#450] [Q&A] [answered]
Hello there,
I'm trying to create JWE as shown in the example from the documentation, but I receive the following error in Node.js:
"throw new TypeError(`${is_key_like_js_1.types.join(' or ')} **instances for symmetric algorithms must be of type "secret"`);"**

Could you please share an example of "secretKey" creation?

I tried all the variants from the following page from the DOC, but the same error is thrown: https://github.com/panva/jose/blob/main/docs/types/types.KeyLike.md

```
const jwt = await new jose.EncryptJWT({ 'urn:example:claim': true })
  .setProtectedHeader({ alg: 'dir', enc: 'A256GCM' })
  .setIssuedAt()
  .setIssuer('urn:example:issuer')
  .setAudience('urn:example:audience')
  .setExpirationTime('2h')
  .encrypt(secretKey)

console.log(jwt)
```

**Comment by panva** [comment, marked as answer]
There is an API that aids to generate secrets of the right type and bit size for a particular algorithm.

https://github.com/panva/jose/blob/v4.9.2/docs/functions/key_generate_secret.generateSecret.md#readme

**Reply by panva** [reply]
```js
const secret = await jose.generateSecret('A256GCM')
console.log(secret)
```

**Reply by Hleb-Haiduk-Software** [reply]
Thank you for your answer, it works, but now I can't understand where my private and public secret keys from the .env file should be used and how I will be able to verify that JWE later without my keys?

JWE creation:
```
  const jwe = await new jose.EncryptJWT({ 'urn:example:claim': true })
    .setProtectedHeader({ alg: 'dir', enc: 'A256GCM' })
    .setIssuedAt()
    .setIssuer('urn:example:issuer')
    .setAudience('urn:example:audience')
    .setExpirationTime('2h')
    .encrypt(await jose.generateSecret('A256GCM'));
```

**Reply by panva** [reply]
I believe in most runtimes the secret generated is just a Uint8Array instance. So that's what you need to be able to (de)serialize.

In any case, whatever the generate APIs produce are just standard language or runtime builtins. How to work with those? Check the respective "thing"'s documentation
